14-tidyr.Rmd

```diff
@@ -0,0 +1,154 @@
+ 
+###Getting started
+First install the packages if you haven't already done so (you probably installed dplyr in the previous lesson):
+```{r,eval=FALSE}
+install.packages("tidyr")
+install.packages("dplyr")
+```
+
+Load the packages
+```{r,message=FALSE}
+library("tidyr")
+library("dplyr")
+```
+
+First, lets look at the structure of our gapminder dataframe:
+```{r}
+str(gapminder)
+```
+
+It is somewhere in between the purely 'long' and 'wide' format. We have 3 "ID variables" (`continent`, `country`, `year`) and 3 "Observation variables" (`pop`,`lifeExp`,`gdpPercap`). I usually prefer my data in this format in most cases despite not having ALL observations in 1 column. Since all 3 observation variables have different units, there are not too many operations that would need us to stretch out this dataframe any longer (i.e. 4 ID variables and 1 Observation variable). Many or R's vector based operations like oav(), lm(), or glm() will be easiest in this intermediate format (i.e. long format for each dependent variable, but not the longest format).
+
+However, converting to the pure 'long' format is the easiest way to convert this data to the pure 'wide' format. Manually entering the data into a spreadsheet is more intuitive in the wide format and some plotting functions actually work better in the wide format.
+
+Before we really start, let's re-organize the ID and Observations variables so that they are on order (this will make more sense later). We're swapping `pop` and `continent`
+
+
+```{r}
+gapminder <- gapminder[,c(1,2,4,3,5,6)]
+str(gapminder)
+
+```
+
+
+#### From intermediate to long format with gather()
+The `tidyr` function `gather()` will 'gather' your observation variables into a single variable.
+
+```{r}
+gap_long <- gapminder %>% gather(obs_type,obs_values,pop,lifeExp,gdpPercap)
+str(gap_long)
+```
+
+Here we have used piping syntax which is similar to what we were doing in the previous lesson with dplyr. In fact, these are compatible and you can use a mix of tidyr and dplyr functions by piping them together
+
+Inside `gather()` we first name the new column for the new ID variable (`obs_type`), the name for the new amalgamated observation variable (`obs_value`), then the names of the old observation variable (`pop`,`lifeExp`,`gdpPercap`). Gather also allows the alternative syntax of using the `-` symbol to identify which variables are not to be gathered (i.e. ID variables)
+
+```{r}
+gap_long <- gapminder %>% gather(obs_type,obs_values,-continent,-country,-year)
+str(gap_long)
+```
+
+That may seem trivial with this particular dataframe, but sometimes you have 1 ID variable and 40 Observation variables. The flexibility is a huge time saver!
+
+> #### Challenge 1 {.challenge}
+>
+> Using `gap_long`, can you calculate the mean life expectancy, population, and gdpPercap for each continent?
+>**Hint:** use the `group_by()` and `summarize()` functions we learned in the dplyr lesson
```

Comment:
dplyr -> `dplyr` 
Important to be consistent in the use of monospaced code font